desci-server/src/controllers/users/associateWallet.ts

```diff
@@ -196,6 +212,68 @@ export const associateWallet = async (req: Request, res: Response, next: NextFun
   }
 };
 
+export const walletLogin = async (req: Request, res: Response, next: NextFunction) => {
+  const logger = parentLogger.child({
+    module: 'USERS::WalletLoginController',
+    user: (req as any).user,
+    body: req.body,
+  });
+  // try {
+  const { message: siweMessage, signature, dev } = req.body;
+  logger.info('WALLET LOGIN');
+
+  if (!siweMessage) {
+    res.status(422).json({ message: 'Expected prepareMessage object as body.' });
+    return;
+  }
+
+  const message = new SiweMessage(siweMessage);
+  const fields = await message.validate(signature);
+  const siweNonce = await extractTokenFromCookie(req, 'siwe');
+  const account = getAddress(fields.address);
+
+  logger.info({ siweNonce }, 'SIWE NONCE');
+  if (fields.nonce !== siweNonce) {
```

Comment:
This doesn't necessary means it's insecure because, because if you hijack a signed messaged and try to replay the auth you'd still have no access to the updated siwe nonce as the except you hijack immediately on the client side as the user signs in and before the siwe nonce httpOnly cookie expires ( which is 1 Minute)

Reply:
The only thing that'll be an added layer security value here will be adding a 2FA for wallet logins not storing siwe in the database